docs/glossary.md

```diff
@@ -31,3 +31,19 @@
   - Defending units can only participate in this phase if `Submarines Defending May Submerge Or Retreat` property is set.
   - `Sub Retreat Before Battle` defines if this phase occurs before the First Strike Phase or at the end of the Battle Round.
 - **Submerge Subs Phase**: Another name for **Evasion Phase** and generally used when only submarines have `canEvade` and `Submersible Subs` is set.
+
+##Development Terms
+- **Branch**: A Branch is a copy of the main codebase.  It is used to keep changes separate until they are completed or in a stable state.  
+- **Checkout**: When you Checkout a branch, that is where commits are stored.  
+- **Clone**: Cloning is when you take your repository and copy it to your local machine to work on.  
+- **Commit**: A Commit is a list of changes being made to a repository.  
+- **Fetch**: When you run a fetch command, your IDE is updating information in your local project.  
```

Comment:
The information is updated in git, specifically this information is stored in the `.git` folder. One can think of a 'fetch' as a remote refresh. What you are doing is grabbing the full set of all commits and the full set of branch pointers from a remote repository and downloading that.